src/discljord/formatting.clj

```diff
@@ -0,0 +1,93 @@
+(ns discljord.formatting
+  "Contains utility functions to help with Discord message formatting.")
+
+(def user-mention #"<@!?(?<id>\d+)>")
+
+(def role-mention #"<@&(?<id>\d+)>")
+
+(def channel-mention #"<#(?<id>\d+)>")
+
+(def emoji-mention #"<(?<animated>a)?:(?<name>):(?<id>\d+)>")
```

Comment:
This doesn't quite work, as it doesn't include what the name can be. This should probably be changed to `#"<(?<animated>a)?:(?<name>\S+):(?<id>\d+)>"`

Reply:
you're right, I tried to find the legal names but forgot to add a pattern. I think it's just alphanumeric, but I'm not 100% sure yet.

Reply:
I've tried most special characters I can think of, I am pretty certain now that only `[A-Za-z0-9_]` is allowed.

Reply:
Others over at the Discord API server just confirmed this and other apps seem to use the same regex.